schematic/store/synapse.py

```diff
@@ -577,6 +581,47 @@ def format_annotations(self, se, row, entityId, useSchemaLabel, hideBlanks):
                 annos[anno_k] = anno_v
         return annos
 
+    def format_manifest_annotations(self, manifest, manifest_synapse_id):
+        '''
+        Set annotations for the manifest (as a whole) so they can be applied to the manifest table or csv.
+        For now just getting the Component.
+        '''
+        
+        entity = self.syn.get(manifest_synapse_id, downloadFile=False)
+        is_file = entity.concreteType.endswith(".FileEntity")
+        is_table = entity.concreteType.endswith(".TableEntity")
+
+        if is_file:
+            # Get file metadata
+            metadata = self.getFileAnnotations(manifest_synapse_id)
+
+            # Gather component information
+            component = manifest['Component'].unique()
+            
+            # Double check that only a single component is listed, else raise an error.
+            try:
+                len(component) == 1
+            except ValueError as err:
+                raise ValueError(
+                    f"Manifest has more than one component. Please check manifest and resubmit."
+                ) from err
+
+            # Add component to metadata
+            metadata['Component'] = component[0]
+        
+        elif is_table:
+            # Get table metadata
+            metadata = self.getTableAnnotations(manifest_synapse_id)
+        
+        # Get annotations
+        annos = self.syn.get_annotations(manifest_synapse_id)
+
+        # Add metadata to the annotations
+        for annos_k, annos_v in metadata.items():
+            annos[annos_k] = annos_v
+
+        return annos
+
     def associateMetadataWithFiles(
         self, metadataManifestPath: str, datasetId: str, manifest_record_type: str, 
```

Comment:
@mialy-defelice could you add a default option for manifest_record_type? We can go with "both" for now. This would avoid problems with functions that [call](https://github.com/Sage-Bionetworks/schematic/blob/a9301999d407b10caa7370478ae96cff32f32472/schematic/store/synapse.py#L393) this function but don't provide this record type parameter.